packages/dds/sequence/src/intervals/intervalUtils.ts

```diff
@@ -0,0 +1,248 @@
+	 * This should just be the refSeq on the op that modified the interval, which should be available via other means.
+	 * At the time of writing, it's not plumbed through to the reconnect/rebase code, however, which does need it.
+	 */
+	sequenceNumber: number;
+	/** Start position of the interval */
+	start: number;
+	/** End position of the interval */
+	end: number;
+	/** Interval type to create */
+	intervalType: IntervalType;
+	stickiness?: IntervalStickiness;
+	/** Any properties the interval has */
+	properties?: PropertySet;
+}
+
+export interface ISerializableInterval extends IInterval {
+	/** Serializable bag of properties associated with the interval. */
+	properties: PropertySet;
+	/** @internal */
+	propertyManager: PropertiesManager;
+	/** @internal */
+	serialize(): ISerializedInterval;
+	/** @internal */
+	addProperties(
+		props: PropertySet,
+		collaborating?: boolean,
+		seq?: number,
+	): PropertySet | undefined;
+	/**
+	 * Gets the id associated with this interval.
+	 * When the interval is used as part of an interval collection, this id can be used to modify or remove the
+	 * interval.
+	 * @remarks - This signature includes `undefined` strictly for backwards-compatibility reasons, as older versions
+	 * of Fluid didn't always write interval ids.
+	 */
+	getIntervalId(): string | undefined;
+}
+
+/**
+ * Represents a change that should be applied to an existing interval.
+ * Changes can modify any of start/end/properties, with `undefined` signifying no change should be made.
+ * @internal
+ */
+export type SerializedIntervalDelta = Omit<ISerializedInterval, "start" | "end" | "properties"> &
+	Partial<Pick<ISerializedInterval, "start" | "end" | "properties">>;
+
+/**
+ * A size optimization to avoid redundantly storing keys when serializing intervals
+ * as JSON for summaries.
+ *
+ * Intervals are of the format:
+ *
+ * [start, end, sequenceNumber, intervalType, properties, stickiness?]
+ */
+export type CompressedSerializedInterval =
+	| [number, number, number, IntervalType, PropertySet, IntervalStickiness]
+	| [number, number, number, IntervalType, PropertySet];
+
+/**
+ * @sealed
+ */
+export interface IIntervalHelpers<TInterval extends ISerializableInterval> {
+	compareEnds(a: TInterval, b: TInterval): number;
+	compareStarts?(a: TInterval, b: TInterval): number;
+	/**
+	 *
+	 * @param label - label of the interval collection this interval is being added to. This parameter is
+	 * irrelevant for transient intervals.
+	 * @param start - numerical start position of the interval
+	 * @param end - numerical end position of the interval
+	 * @param client - client creating the interval
+	 * @param intervalType - Type of interval to create. Default is SlideOnRemove
+	 * @param op - If this create came from a remote client, op that created it. Default is undefined (i.e. local)
+	 * @param fromSnapshot - If this create came from loading a snapshot. Default is false.
+	 */
```

Comment:
nit: missing @param tag for stickiness

Reply:
It appears like the tag was missing in the original implementation. Have added in the latest commit.